pkg/appgw/types.go

```diff
@@ -0,0 +1,20 @@
+package appgw
+
+import (
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/environment"
+	"k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+
+	mtv1 "github.com/Azure/application-gateway-kubernetes-ingress/pkg/apis/azureingressmanagedtarget/v1"
+	ptv1 "github.com/Azure/application-gateway-kubernetes-ingress/pkg/apis/azureingressprohibitedtarget/v1"
+)
+
+// KubernetesResources holds the structs we have fetches from Kubernetes + environment, based on which
+// we will construct App Gateway config.
+type KubernetesResources struct {
```

Comment:
`BuilderContext`/`BuilderResources` ?

Reply:
@akshaysngupta  I took a more liberal approach with the naming here - decided to scope it to ConfigBuilder -- so it is clear which this builder this is for.
It does get a bit more verbose -- `ConfigBuilderContext` - but it actually semantically is unambiguous - this is the context within/with which we create the App Gwy config.
Thanks for the great suggestion!